### ISSUE TITLE
Hitting result max when using too many query selectors

### ISSUE THREAD

**Issue by jasonsims** [jsforce/jsforce#36]
I'm running into an issue where a max of only 200 records are returned if several query selectors are being used. This is an issue for me since I'm selecting all attributes on the corresponding model within the ORM so that I don't get back every field from salesforce. The issue can be reproduced by using the all selector `'*'`. For example, assuming the `Account` object has 450 records:  

``` js
// This query will return all 450 records.
conn.sobject('Account')
  .select('Id')
  .execute(function(err, res) {
    console.log('LEN: ' + res.length)
  })

// However, this query will only return 200 records
conn.sobject('Account')
  .select('*')
  .execute(function(err, res) {
    console.log('LEN: ' + res.length)
  })
```

I've ran the query directly against  SFDC API using all of the model attributes and it returns all records for me so something may be causing this issue within jsforce. I also checked on what [lib/connection.js](https://github.com/jsforce/jsforce/blob/master/lib/connection.js) is sending to the API and it seems to build the HTTP request just fine. It's possible that something is up with the post processing of the response body but I'm not entirely sure.

**Comment by turakvlad** [comment]
Hello, @stomita!

As I understood this query will return all accounts (even if we have dozens of thousands)? 

```
const accounts = await connection.sobject('Account').find()
```

And this query will return all account's contacts:

```
const contacts = await connection.sobject('Contact').find({
  AccountId: 'account_id_here'
}, '*')
```

Is that correct? 

In [this PR](https://github.com/jsforce/jsforce/pull/37) I can see that it should return all the records _till `maxFetch` num if `autoFetch` option is specified_. But how should I specify `maxFetch` and `autoFetch` values? Where should they be set? Should I set them or everything will work under the hood?

Thanks.

**Comment by mcbergsma** [comment]
I just ran into this issue as well using the event driven style. I tried @turakvlad 's suggestion of appending `,'*'` to my `find`, but it didn't work. I couldn't find any documentation on how to implement it, but it turns out this works.

```
req.conn.sobject('customObj')
  .find({stuff})
  .autoFetch(true)
  .maxFetch(100000)
  .execute((err, results) => .... )
```

I don't have more than 10K records to fetch, but setting `maxFetch` alone to a large number didn't make any difference, removing it and only using `autoFetch` was enough for my needs, but I added the `maxFetch` option anyway for a bit of future-proofing since I assume it will default to 10K.
